adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -523,13 +600,15 @@ protected void handleRemoteReceiverOpen(final ProtonConnection conn, final Proto
      * </ul>
      * 
      * @param ctx The context for the message.
+     * @param traceSamplingPriority The sampling priority to be applied on the <em>OpenTracing</em> span
+     *                              created for this operation.
      * @return A future indicating the outcome of processing the message.
      *         The future will succeed if the message has been processed successfully, otherwise it
      *         will fail with a {@link ServiceInvocationException}.
      */
-    protected Future<ProtonDelivery> onMessageReceived(final AmqpContext ctx) {
+    protected Future<ProtonDelivery> onMessageReceived(final AmqpContext ctx, final OptionalInt traceSamplingPriority) {
```

Comment:
why don't we put the trace sampling priority into the `AmqpContext` itself?

Reply:
@calohmn any opinion on this?

Reply:
I agree, I've amended the commit accordingly.